### ISSUE TITLE
Ability to modify TC in report

### ISSUE THREAD

**Issue by maehler** [hydra-genetics/reports#18]
A nice feature would be the ability to modify the TC on the fly inside the report. In cases where the TC estimate is off, the results can be severly affected by it. It is apparent that it is wrong, but it is not apparent what CNVs would have been called had the TC been more accurate. Modifying the TC should then essentially move the baseline in the log<sub>2</sub> ratio plots.

A consequence of this is that the copy numbers in the table of calls should be calculated on the fly as well, based on the new log<sub>2</sub> ratios. A potential problem with this is that the table contains filtered calls from the CNV calling, meaning that any new CNVs detected when modifying the TC will not readily show up among the filtered results. Somehow the thresholds for amplifications and losses have to be integrated into the report, both when it comes to what is displayed in the filtered table, but also how the CNVs are classified (i.e. DEL, DUP, COPY_NORMAL).

@jonca79: please add anything that I might have overlooked related to this.

**Comment by maehler** [comment]
Quite a while since this issue was opened, but now I'm slowly getting into this.

Back when I posted the issue I had misunderstood what this would actually do. It won't affect the baseline, but rather modify the log<sub>2</sub> ratios in order to compensate for an impure sample.

At the moment I will focus on visualising this only in the plot, and not try to modify the results table. There might be value in having the results table represent the values that were generated by the underlying tools in order to have some frame of reference. It will have to be made very clear in the report that this is the case. The main issue with modifying the values in the table is that in order to be complete, it would have to be able to inject new variants in addition to modifying existing variants. This could become quite messy, so I think this should be considered out of scope at this time.

**Comment by jonca79** [comment]
I totally agree with only changing the log2 ratios in the plot. Maybe you could do an export function in the future.
One thing that would add much interpretation value is to add a secondary Y-axis showing copy numbers when the data is TC-modified. Alternatively, this is a separate feature that should always be there.

**Comment by maehler** [comment]
Yes, I agree that a secondary y-axis would be nice. If this is displayed by default, even if the TC hasn't been applied/modified, I'm a bit worried that people might get confused as to why the values are not the same as in the table. This might not be a huge issue though since it anyways has to be made clear what is being displayed in the plot given the settings that have been applied.

**Comment by maehler** [comment]
I added the secondary y-axis to #90.
